src/main/java/com/researchspace/webapp/integrations/fieldmark/FieldmarkApiController.java

```diff
@@ -0,0 +1,156 @@
+package com.researchspace.webapp.integrations.fieldmark;
+
+import static com.researchspace.service.fieldmark.impl.FieldmarkToRSpaceApiConverter.createContainerRequest;
+import static com.researchspace.service.fieldmark.impl.FieldmarkToRSpaceApiConverter.createSampleRequest;
+import static com.researchspace.service.fieldmark.impl.FieldmarkToRSpaceApiConverter.createSampleTemplateRequest;
+
+import com.researchspace.api.v1.FieldmarkApi;
+import com.researchspace.api.v1.controller.ApiController;
+import com.researchspace.api.v1.controller.BaseApiController;
+import com.researchspace.api.v1.controller.ContainersApiController;
+import com.researchspace.api.v1.controller.InventoryFilesApiController;
+import com.researchspace.api.v1.controller.InventoryFilesApiController.ApiInventoryFilePost;
+import com.researchspace.api.v1.controller.SampleTemplatesApiController;
+import com.researchspace.api.v1.controller.SamplesApiController;
+import com.researchspace.api.v1.model.ApiContainer;
+import com.researchspace.api.v1.model.ApiField.ApiFieldType;
+import com.researchspace.api.v1.model.ApiSampleField;
+import com.researchspace.api.v1.model.ApiSampleTemplate;
+import com.researchspace.api.v1.model.ApiSampleTemplatePost;
+import com.researchspace.api.v1.model.ApiSampleWithFullSubSamples;
+import com.researchspace.fieldmark.model.FieldmarkMultipartFile;
+import com.researchspace.fieldmark.model.FieldmarkNotebook;
+import com.researchspace.fieldmark.model.utils.FieldmarkFileExtractor;
+import com.researchspace.model.User;
+import com.researchspace.model.dtos.FieldmarkNotebookDTO;
+import com.researchspace.model.dtos.FieldmarkRecordDTO;
+import com.researchspace.service.fieldmark.FieldmarkServiceClientAdapter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.BindException;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.client.HttpServerErrorException;
+
+@Slf4j
+@ApiController
+public class FieldmarkApiController extends BaseApiController implements FieldmarkApi {
+
+  private @Autowired SampleTemplatesApiController sampleTemplatesApiController;
```

Comment:
It's a bit strange to be depending on one Controller from another and creating Inventory items via the controller rather than via the Service classes. Is there a reason for this? I know in a lot of cases we do much more work in the Controllers than we should be, which leads to the problem where the Service classes become less useful and reusable. By depending on Controllers here were are coupling parts of the system with e.g. the response the client/front-end gets from `SampleTemplatesApiController` endpoints.

For example, rather than creating the Rspace objects via these 4 Controllers, it would make more sense to call the Service classes e.g. `SampleApiManager.createNewSampleTemplate` rather than `SampleTemplatesApiController.createNewSampleTemplate`.

If you are relying on something which is currently done in the controller methods rather than the services then maybe we should create a tech debt ticket to move the logic from the controller to the service classes in these cases? It's something we should probably take a look at across the code-base at some point too.

Reply:
Yes correct, the creation of the samples and sample template is delegated to the controllers since they do quite a lot of work in diufferent ways: calling different services and actually doing quite a lot of logic (validatio, building links for the inventory and updating samples with the number of subsamples).
So with @mKowalski256 we decided to actually make the controller as peerer to be able to call each other.

Reply:
Yeah I can see why you've done it this way then. But we do have a problem across the codebase with doing too much inside Controller classes. I will create a ticket for us to look into refactoring the Controllers and Services (not specific to this case but on the codebase as a whole) .